getting-started/bookshelf/main_test.go

```diff
@@ -16,47 +16,57 @@ package main
 
 import (
 	"bytes"
+	"context"
 	"fmt"
+	"io/ioutil"
+	"log"
 	"mime/multipart"
 	"net/http/httptest"
 	"os"
 	"strings"
 	"testing"
 
-	"github.com/GoogleCloudPlatform/golang-samples/getting-started/bookshelf"
-	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
-	"github.com/GoogleCloudPlatform/golang-samples/internal/webtest"
+	"github.com/GoogleCloudPlatform/golang-samples/getting-started/bookshelf/internal/webtest"
 )
 
-var wt *webtest.W
+var (
+	wt *webtest.W
+	b  *Bookshelf
+)
 
 func TestMain(m *testing.M) {
+	projectID := os.Getenv("GOLANG_SAMPLES_FIRESTORE_PROJECT")
+	if projectID == "" {
+		fmt.Fprintln(os.Stderr, "GOLANG_SAMPLES_FIRESTORE_PROJECT not set. Skipping.")
+		return
+	}
+
+	// Don't log anything during testing.
+	log.SetOutput(ioutil.Discard)
```

Comment:
maybe capture output in a buffer and print only if tests failed? probably doesn't matter

Reply:
Will leave as-is, but we can change if if/when this fails and we need debugging.